ADC/Core/Inc/macros.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * macros.h
+ *
+ *  Created on: Mar 6, 2020
+ *      Author: rodrigocelmer
+ */
+
+#ifndef INC_MACROS_H_
+#define INC_MACROS_H_
+
+//--- Temperature sensor calibration value address - from Reference Manual
+//we use them to calculate the temperature read by the ADC
+#define TEMP130_CAL_ADDR	((uint16_t*) ((uint32_t) 0x1FF8007E))
+#define TEMP30_CAL_ADDR		((uint16_t*) ((uint32_t) 0x1FF8007A))
+#define VDD_CALIB			((uint16_t) (300))
+#define VDD_APPLI			((uint16_t) (330))
+
+//--- Internal voltage reference value address - from Reference Manual
+//we use it to calculate the Vref read by the ADC
+#define VREFINT_CAL		((uint16_t*) ((uint32_t) 0x1FF80078))
+
+//--- Time management defines
+/*
+ * Here we define constant values for Wait(uint16_t) function
+ * System clock is running at 2.1 MHz
+ * Ex.: 10 uS / (1 / 2.1 MHz) = 21 system clock cycles to wait for 10 uS
+ */
+#define TIME_10uSEC		21	//21 system clock cycles to wait for 10uS
+
```

Comment:
How do you know how many clock cycles it takes for the processor run the following steps?
```
while(time)
{
       time--;
}
```
How many clock cycles does it takes to check of time is true or false?
How many clock cycles does it takes to decrement time by one?

Turns out that you don't really know that for two reasons:
1. You will need to check which assembly commands the processor is running in order to do those operations;
2. Even if you know that, those commands may change based on the compiler you are using.

Relying on clock cycles to generate a delay is not a good way to do it, unless you really know that the output of your compiler is always the same.

Reply:
yes, yes, I know that this delay takes more than the time I need, but I'm not using any of the timers here and I need that delay to initialize the temperature sensor properly (if I don't use this delay, the temperature's not correctly measured). Also, I don't have a flag to know when the temperature sensor is ready to be used, so that was the best but the shitty option I could think of since I don't have a timer enabled

Reply:
Ok. So it is better you to explain that you are only waiting a few clock cycles until the temperature sensor is ready than make an invalid calculation. No problem with that.